src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,199 @@
+                    PowerShellWorkerStrings.LoadingDurableSDK, PowerShellWorkerStrings.ExternalDurableSDKName));
+
+                var results = _pwsh.AddCommand(Utils.ImportModuleCmdletInfo)
+                    .AddParameter("FullyQualifiedName", PowerShellWorkerStrings.ExternalDurableSDKName)
+                    .AddParameter("ErrorAction", ActionPreference.Stop)
+                    .AddParameter("PassThru")
+                    .InvokeAndClearCommands<PSModuleInfo>();
+
+                var numResults = results.Count();
+                var numExpectedResults = 1;
+                if (numResults != numExpectedResults)
+                {
+                    _logger.Log(isUserOnlyLog: false, LogLevel.Warning, string.Format(
+                               PowerShellWorkerStrings.UnexpectedResultCount, "Import Durable SDK", numExpectedResults, numResults));
+                }
+
+                // log that import succeeded
+                var moduleInfo = results[0];
+                _logger.Log(isUserOnlyLog: false, LogLevel.Trace, String.Format(
+                    PowerShellWorkerStrings.ImportSucceeded, moduleInfo.Name, moduleInfo.Version));
+
+                importSucceeded = true;
+            }
+            catch (Exception e)
+            {
+                // If an error ocurred, we try to log the exception.
+                var errorMessage = e.ToString();
+
+                // If a PowerShell error record is available through Get-Error, we log that instead.
+                if (e.InnerException is IContainsErrorRecord inner)
+                {
+                    errorMessage = _errorRecordFormatter.Format(inner.ErrorRecord);
+
+                }
+                _logger.Log(isUserOnlyLog: true, LogLevel.Error, string.Format(
+                    PowerShellWorkerStrings.ErrorImportingDurableSDK,
+                    PowerShellWorkerStrings.ExternalDurableSDKName, errorMessage));
+
+            }
+            return importSucceeded;
+
+        }
+
+        public void tryEnablingExternalDurableSDK()
+        {
+            // Search for the external DF SDK in the available modules
+            var matchingModules = _pwsh.AddCommand(Utils.GetModuleCmdletInfo)
+                .AddParameter("ListAvailable")
+                .AddParameter("FullyQualifiedName", PowerShellWorkerStrings.ExternalDurableSDKName)
+                .InvokeAndClearCommands<PSModuleInfo>();
+
+            // If we get at least one result, we attempt to load it
+            var numCandidates = matchingModules.Count();
+            if (numCandidates > 0)
+            {
+                // try to import the external DF SDK
+                _usesExternalDurableSDK = tryImportingDurableSDK();
+            }
+            else
+            {
+                // Log that the module was not found in worker path
+                _logger.Log(isUserOnlyLog: false, LogLevel.Trace, string.Format(
```

Comment:
Should we log here the values of `$env:PSModulePath`? If we decide to do it, please log the value to Kusto.